### ISSUE TITLE
No any logs of Kafka Consumer misconfiguration

### ISSUE THREAD

**Issue by merlin-zaraza** [spring-cloud/spring-cloud-stream#3024]
I've faced with next issue:
```
2024-10-24T16:08:32.794+03:00 ERROR 77620 --- [processor] [   scheduling-1] o.s.cloud.stream.binding.BindingService  : Failed to create consumer binding; retrying in 30 seconds

org.springframework.cloud.stream.binder.BinderException: Cannot initialize binder checking the topic (my-topic):
	at org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner.getPartitionsForTopic(KafkaTopicProvisioner.java:685) ~[spring-cloud-stream-binder-kafka-core-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner.getPartitionInfoForConsumer(KafkaTopicProvisioner.java:585) ~[spring-cloud-stream-binder-kafka-core-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner.getListenedPartitions(KafkaTopicProvisioner.java:557) ~[spring-cloud-stream-binder-kafka-core-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.createConsumerEndpoint(KafkaMessageChannelBinder.java:606) ~[spring-cloud-stream-binder-kafka-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.createConsumerEndpoint(KafkaMessageChannelBinder.java:168) ~[spring-cloud-stream-binder-kafka-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.doBindConsumer(AbstractMessageChannelBinder.java:529) ~[spring-cloud-stream-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.doBindConsumer(AbstractMessageChannelBinder.java:103) ~[spring-cloud-stream-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binder.AbstractBinder.bindConsumer(AbstractBinder.java:144) ~[spring-cloud-stream-4.1.3.jar:4.1.3]
	at org.springframework.cloud.stream.binding.BindingService.lambda$rescheduleConsumerBinding$1(BindingService.java:211) ~[spring-cloud-stream-4.1.3.jar:4.1.3]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-6.1.3.jar:6.1.3]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:840) ~[na:na]
Caused by: java.lang.RuntimeException: Failed to obtain partition information for the topic my-topic
	at org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner.lambda$getPartitionsForTopic$9(KafkaTopicProvisioner.java:658) ~[spring-cloud-stream-binder-kafka-core-4.1.3.jar:4.1.3]
	at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:335) ~[spring-retry-2.0.5.jar:na]
	at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:211) ~[spring-retry-2.0.5.jar:na]
	at org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner.getPartitionsForTopic(KafkaTopicProvisioner.java:622) ~[spring-cloud-stream-binder-kafka-core-4.1.3.jar:4.1.3]
	... 16 common frames omitted
```

And after few hours of unsuccessful investigation I've put breakpoint on `KafkaTopicProvisioner.getPartitionInfoForConsumer` to debug my consumer configuration. What I've discovered that there are no any logs at [this try-catch block](https://github.com/spring-cloud/spring-cloud-stream/blob/12468dcd0a06ee6ea87d4e7488bdb378d181e823/binders/kafka-binder/spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java#L670).
It's not a good practice to silence exceptions. It would be nice to log all exceptions to get more details on root of error cause. In my case that was a lack of `KafkaAvroDeserializer` class in class path.

**Comment by sobychacko** [comment]
Hi, Thanks for this report. Did you create this same [Stack Overflow thread](https://stackoverflow.com/questions/79121257/topic-partition-with-spring-cloud-stream-can-not-consumer-topic)? It looks similar. 
We will address this soon.

**Comment by sobychacko** [comment]
A fix is added for logging. Also, back-ported to the `4.1.x` branch. Please take a look and let us know if it helps in situations like the one you raised. Then, we can close the issue. Thanks!

**Comment by merlin-zaraza** [comment]
It's definitely helpful. Thanks a lot @sobychacko

**Comment by sobychacko** [comment]
Closing the issue via the commit mentioned above.

**Comment by alevezendanBT** [comment]
Hello @merlin-zaraza. I'm facing the same issue, did you manage to fix it? Thanks.

**Comment by merlin-zaraza** [comment]
> Hello @merlin-zaraza. I'm facing the same issue, did you manage to fix it? Thanks.

Yes, in my case there was no `KafkaAvroDeserializer` in classpath (e.g. no library in my pom file).
It might be different in your case. Please see logs with fixed version (4.2.0 should contain fix).
